### PR TITLE
Fix switch/case label error, tweak status handling

### PR DESCRIPTION
**Pull request by jajik**
Fix a compilation error (#82) introduced with the new code style. The PR also refactors duplicated functionality.

### CODE REVIEW

**Comment by jajik** [comment]
I added a second commit that makes the pipeline to build with both, clang and gcc. It might prevent some compilation errors as the one in #82.

(Without the 1ebdbfe11b137e8c9e179a28ddf88c5fde1f0f05, the job would fail. See https://github.com/jajik/mod_proxy_cluster/actions/runs/5011990175/jobs/8983418382)

**Comment by rhusar** [comment]
> Note that the CI reports `Waiting for status to be reported` for `cmake-fedora-latest` job. This is due to the change in CI, where the job of that name now runs in two flavours – the old name is no longer used, but PR was opened while it existed and it stayed there.
> 
> TL;DR it should be ignored.

That's correct - I will just change the required checks set it once we merge this.

**Comment by rhusar** [comment]
> > Note that the CI reports `Waiting for status to be reported` for `cmake-fedora-latest` job. This is due to the change in CI, where the job of that name now runs in two flavours – the old name is no longer used, but PR was opened while it existed and it stayed there.
> > TL;DR it should be ignored.
> 
> That's correct - I will just change the required checks set once we merge this.

Required checks are now updated.
